moesifmiddleware.go

```diff
@@ -375,3 +391,21 @@ func sendEvent(request *http.Request, response MoesifResponseRecorder, rspBuffer
 	// Send Event To Moesif
 	sendMoesifAsync(request, reqTime, requestHeader, apiVersion, reqBody, &reqEncoding, rspTime, response.status, responseHeader, respBody, &respEncoding, userId, companyId, &sessionToken, metadata, &direction)
 }
+
+// teeBody reads all of b to memory and then returns two equivalent
+// ReadClosers yielding the same bytes.
+// It returns an error if the initial slurp of all bytes fails.
+func teeBody(b io.ReadCloser) (r1, r2 io.ReadCloser, err error) {
+	if b == nil || b == http.NoBody {
+		// No copying needed
+		return http.NoBody, http.NoBody, nil
+	}
+	var buf bytes.Buffer
+	if _, err = buf.ReadFrom(b); err != nil {
+		return nil, b, err
+	}
+	if err = b.Close(); err != nil {
+		return nil, b, err
+	}
+	return io.NopCloser(&buf), io.NopCloser(bytes.NewReader(buf.Bytes())), nil
```

Comment:
@bakennedy the correct package here is ioutil which has NopCloser function. It should be `ioutil.NopCloser` instead of `io.NopCloser`.  This version won't get installed as it'll throw undefined error on `io.NopCloser`.